cli/azd/pkg/infra/provisioning/bicep/bicep_provider.go

```diff
@@ -645,31 +645,42 @@ func latestCompletedDeployment(
 	return nil, fmt.Errorf("no deployments found for environment %s", envName)
 }
 
-// resourceGroupsFromDeployment returns the names of all the unique set of resource group name names resource groups from
-//
-//	the OutputResources section of a ARM deployment.
-func resourceGroupsFromDeployment(deployment *armresources.DeploymentExtended) []string {
-
+// resourceGroupsToDelete collects the resource groups from an existing deployment which should be removed as part of a
+// destroy operation.
+func resourceGroupsToDelete(deployment *armresources.DeploymentExtended) []string {
```

Comment:
Does this handle rg-level deployments as well? What is the resulting behavior there?

Reply:
This works if the resource group scoped deployment itself did nested deployments into that resource group (we'll see references to the nested deployments, for example here's the output of `az deployment group show` for the TODO nodejs app ported to use a resource group: https://gist.github.com/ellismg/cafe52b6142776b6400e5dbec30663b8